apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -975,15 +958,34 @@ export class InstallManager {
    */
   private _pushConfigurationArgs(args: string[], options: IInstallManagerOptions): void {
     if (this._rushConfiguration.packageManager === 'npm') {
-      args.push('--no-optional');
+      if (semver.lt(this._rushConfiguration.packageManagerToolVersion, '5.0.0')) {
+        // NOTE:
```

Comment:
I kept this comment (which was originally added in https://github.com/Microsoft/web-build-tools/pull/175), reworded it a bit, and added a couple useful links. lmk if I should change anything else here.